operator/v1/types_ingress.go

```diff
@@ -225,6 +241,65 @@ var (
 	DNSReadyIngressConditionType = "DNSReady"
 )
 
+// Ciphers defines the cipher algorithms used by an IngressController for performing
+// encryption or decryption of network connections. A profile can be specified or a
+// custom strategy of ciphers.
+type Ciphers struct {
+	// Profile is one of three recommended cipher configurations from [1].
+	//
+	// [1] https://wiki.mozilla.org/Security/Server_Side_TLS#Recommended_configurations
+	//
+	// +optional
+	Profile CipherProfile `json:"profile,omitempty"`
+	// CustomStrategy consists of a manually defined list of ciphers with each
+	// cipher separated by a ":".
+	//
+	// +optional
+	CustomStrategy *string `json:"customStrategy,omitempty"`
+}
+
+// CipherProfile is a way to specify a supported cipher profile.
```

Comment:
Is there a way to specify an unsupported cipher profile?

Reply:
No. I have removed the wording. `CipherProfile` is now `Profile` within `SecuritySpec`:
```
// SecuritySpec defines the settings for securing IngressController connections.
type SecuritySpec struct {
	// Profile is one of three possible security configurations from [1]:
	//
	// [1] https://wiki.mozilla.org/Security/Server_Side_TLS#Recommended_configurations
	//
	// If unset, the default is "intermediate".
	//
	// +optional
	Profile *SecurityProfileType `json:"profile,omitempty"`
	// CustomSettings defines the schema for customizing the security settings of
	// IngressController connections.
	//
	// +optional
	CustomSettings *CustomProfileSettings `json:"customSettings,omitempty"`
}
```